src/SmartComponents/AdvisorySystems/AdvisorySystem.test.js

```diff
@@ -49,150 +38,128 @@ const mockState = {
 };
 
 const initStore = () => {
-    const customMiddleWare = () => next => action => {
-        useSelector.mockImplementation(callback => {
-            return callback(mockState);
-        });
-        next(action);
-    };
-
-    const mockStore = configureStore([customMiddleWare]);
+    const mockStore = configureStore([]);
     return mockStore(mockState);
 };
 
-let wrapper;
-let store = initStore();
-// eslint-disable-next-line no-unused-vars
-const rejectedState = { entities: { ...mockState, status: 'rejected', error: { detail: 'test' } },
-    AdvisorySystemsStore: mockState.AdvisorySystemsStore };
+const store = initStore();
 beforeEach(() => {
-    useSelector.mockImplementation(callback => {
-        return callback(mockState);
-    });
-
-    /**
- * This mount snapshot test is failing because of randomly generated key in PF component.
- * Before the snapshot was only shallow and did not display the real virtual DOM.
- * Shallow snapshot has the same result as the mount before.
- */
-    wrapper = mount(<Provider store={store}>
-        <Router> <AdvisorySystems advisoryName={'RHSA-2020:2755'} /></Router>
-    </Provider>);
-});
-
-afterEach(() => {
-    useSelector.mockClear();
-    store.clearActions();
+    render(<ComponentWithContext renderOptions={{ store }}>
+        <AdvisorySystems advisoryName={'RHSA-2020:2755'} />
+    </ComponentWithContext>);
 });
 
+//TODO: find a meaningful way of testing InventoryTable fed module
```

Comment:
I think there won't be a good way to do this. We can with a better mock maybe verify that some props are properly passed in and we might be able to mimic the behaviour of the InventoryTable, but this might be more headache in the long run, especially the last part. 

We should focus more on making sure that everything around works as expected. For example, rather than verifying the `getEntities` by having it called by the `InventoryTable` and expect it to render the systems in the table, call it directly and verify its output matches what the inventory would expect. Or rather than verifying the `onSelect` handler by wanting to click in the table on the checkboxes only verify the `onSelect` handler directly.

Reply:
I agree. Actually, I have started on creating such a mocked component that just triggers events passed as a prop. I am going to create it in a separate PR as I said earlier.

Reply:
@mkholjuraev To elaborate more. I tried this[0]. It only got as far as calling the `getEntities`, because implementing everything else would have required to pretty much reimplement the InventoryTable.

[0] https://github.com/RedHatInsights/compliance-frontend/blob/master/src/SmartComponents/SystemsTable/__mocks__/InventoryTableMock.js#L9

Reply:
I think I could not explain my point, but I believe we are talking about the same thing

Reply:
Not sure I am gonna add anything more than what Sebastian already described. I am against re-implementing InventoryTable just for the sake of testing it. Because of its federated module nature, the best we can do here is to assert on props. For example, I would just check:

- if `hideFilters` has tags and operatingSystem filter set to true (= we want InventoryTable to show only these two filters)
- `columns` receives the function and its return value matches our expectations of the columns we want to see
- that `customFilters` contains the configuration of the custom filters we want to add on top of tags and operatingSystem filters
- `paginationProps` has `isDisabled` set to true if total items are 0
- with `dedicatedAction` we pass the remediation button.

These are just a half of other things we can assert on, I listed them above just for better understanding.

I am also not sure we even want to test that `getEntities` or something else is called. Mainly because this will inevitable involve adding mocked implementation which will be different from and not synced with the real InventoryTable behaviour - this doesn't make sense to test.

Reply:
Yup, I agree. I have previously created similar tests for the ImmutableDevices table that uses the inventory-fed module. Those tests make sure that the correct filters are passed, the correct state is present if there is an error, etc.

Example: https://github.com/RedHatInsights/vulnerability-ui/blob/master/src/Components/SmartComponents/HybridInventory/ImmutableDevicesTab/ImmutableDevices.test.js